behat/symfony2-extension/2.1/features/bootstrap/bootstrap.php

```diff
@@ -4,5 +4,8 @@
 
 // The check is to ensure we don't use .env in production
 if (!isset($_SERVER['APP_ENV'])) {
+    if (!class_exists(Dotenv::class)) {
+        throw new \RuntimeException('You need to define environment variables for configuration or add "symfony/dotenv" as a Composer dependency');
```

Comment:
What about something along the lines of:

`APP_ENV environment variable is not defined. All project's environment variables should be defined or set via a .env file (for which you need to require "symfony/dotenv" as a Composer dependency).`

I'm not very satisfied with the `should be defined or set` part as I'm not sure it's easily understandable. We could probably find something better.

Reply:
What about straightforward; `Loading project environment variables requires symfony/dotenv ~3.4|4.0`?

Reply:
I'd go for `Loading project environment variables from .env file requires symfony/dotenv`, so we don't need to show anything about the version

Reply:
(Whoops, life got busy)

@ro0NL & @Pierstoval: I think the "this is your next step" message should make it clear that Symfony prefers using env variables rather than a .env in production. Thus this ends up being a bit wordy as the user has two options dependant on what they want to do. Given you're more likely to run into this issue while building for production (as dotenv is installed by default in dev) it's worth starting off talking about env variables, rather than directing people straight to dotenv IMO.

The existing message takes inspiration from the other class_exists check in `bin/console` (see below).  I think it's worth keeping the voice between these two messages somewhat consistent, which is why I started with "You need to...".

```php
if (!class_exists(Application::class)) {
     throw new \RuntimeException('You need to add "symfony/framework-bundle" as a Composer dependency.');
}
```

---

How about `APP_ENV environment variable is not defined. You need to define environment variables for configuration or add "symfony/dotenv" as a Composer dependency to load variables from a .env file.`?

And then updating Application class check message to `Application class does not exist. You need to add "symfony/framework-bundle" as a Composer dependency.` Which keeps a "[why the check failed] [how to fix it]" format.

Reply:
@BPScott Your proposal sounds good to me. Can you update the PR?